crates/matrix-sdk/src/sliding_sync/list/mod.rs

```diff
@@ -884,24 +887,55 @@ pub enum SlidingSyncMode {
 
     /// Fully sync all rooms in the background, page by page of `batch_size`,
     /// like `0..=19`, `20..=39`, 40..=59` etc. assuming the `batch_size` is 20.
-    Paging,
+    Paging {
+        /// The batch size.
+        batch_size: u32,
+
+        /// The maximum number of rooms to fetch. `None` to fetch everything
+        /// possible.
+        maximum_number_of_rooms_to_fetch: Option<u32>,
+    },
 
     /// Fully sync all rooms in the background, with a growing window of
     /// `batch_size`, like `0..=19`, `0..=39`, `0..=59` etc. assuming the
     /// `batch_size` is 20.
-    Growing,
+    Growing {
+        /// The batch size.
+        batch_size: u32,
+
+        /// The maximum number of rooms to fetch. `None` to fetch everything
+        /// possible.
+        maximum_number_of_rooms_to_fetch: Option<u32>,
+    },
 }
 
 impl SlidingSyncMode {
     /// Return whether an external caller can modify the sync mode's ranges.
     fn ranges_can_be_modified_by_user(&self) -> bool {
         match self {
             Self::Selective => true,
-            Self::Paging | Self::Growing => false,
+            Self::Paging { .. } | Self::Growing { .. } => false,
         }
```

Comment:
This can be `matches!(self, Self::Selective)`.

Reply:
Oh yeah, right.